### DISCUSSION TITLE
Universal and Native links

### DISCUSSION THREAD

**Discussion by kandrukKV** [WalletConnect/walletconnect-monorepo#4056] [Developer Support]
Hello! I'm using web3modal_v2 and wagmi to connect to my application. When connected, web3modal asks you to select a wallet. How can I get a universal link that is used when connecting? Is this possible?

**Comment by glitch-txs** [comment]
I'm not sure what you mean, could you explain more what you are trying to accomplish?

**Comment by kandrukKV** [comment]
@glitch-txs Hello!

I am using W3modalV2 in Telegram Web App. TWA does not support deep link navigation and W3M doesn't work correctly on mobile devices. However, some wallets app support clicking on universal links. I want to try to do some "hack". When I send a transaction to my wallet, the wallet application (**Mobile** Metamask, **Mobile** Trust Wallet, etc) does not open. If I had a universal link at that moment, then I could programmatically click on it and open the application. How can I get a universal link to a wallet that is connected to W3M?

**Reply by judeProground** [reply]
Hello. It seems like you can use the [Explorer API](https://docs.walletconnect.com/cloud/explorer). The API may return either a mobile_link or universal_link, depending on the platform value. (I'm not an official contributor, by the way.)

[fetchWallets example](https://github.com/WalletConnect/web3modal-react-native/blob/8cd0f48cd6d7f4595c8c89034af3ddec9a49153a/packages/core/src/controllers/ApiController.ts#L212)

**Reply by Vanekchekarmam** [reply]
u found the solution?
